code/parse/sexp_container.cpp

```diff
@@ -1400,6 +1405,87 @@ void sexp_copy_container(int node)
 	}
 }
 
+void sexp_apply_container_filter(int node)
+{
+	Assertion(node != -1, "Apply-container-filter wasn't given a container to apply a filter to. Please report!");
+
+	const char *container_name = CTEXT(node);
+	auto *const p_container = get_sexp_container(container_name);
+
+	if (!p_container) {
+		report_nonexistent_container("Apply-container-filter", container_name);
+		return;
+	}
+	auto &container = *p_container;
+
+	node = CDR(node);
+	Assertion(node != -1, "Apply-container-filter wasn't given a container to use as a filter. Please report!");
+
+	const char *filter_container_name = CTEXT(node);
+	const auto *p_filter_container = get_sexp_container(filter_container_name);
+
+	if (!p_filter_container) {
+		report_nonexistent_container("Apply-container-filter", filter_container_name);
+		return;
+	}
+	if (p_container == p_filter_container) {
```

Comment:
Everything looks good.  I've been told to separate my if blocks for readability, but I don't know if we have that in any official SCP style guide.